components/engine_rocks/src/misc.rs

```diff
@@ -1,108 +1,59 @@
-// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
 
-use crate::rocks;
-use crate::rocks::{Writable, DB};
+use crate::engine::RocksEngine;
+use crate::util;
+use engine_traits::{MiscExt, Range, Result};
 
-use super::Result;
-use super::{IterOption, Iterable};
-use engine_traits::CF_LOCK;
-use tikv_util::keybuilder::KeyBuilder;
-
-use crate::Mutable;
-use rocksdb::WriteBatch;
-impl Mutable for WriteBatch {}
-
-// In our tests, we found that if the batch size is too large, running delete_all_in_range will
-// reduce OLTP QPS by 30% ~ 60%. We found that 32K is a proper choice.
-pub const MAX_DELETE_BATCH_SIZE: usize = 32 * 1024;
-
-pub fn delete_all_in_range(
-    db: &DB,
-    start_key: &[u8],
-    end_key: &[u8],
-    use_delete_range: bool,
-) -> Result<()> {
-    if start_key >= end_key {
-        return Ok(());
+impl MiscExt for RocksEngine {
+    fn is_titan(&self) -> bool {
+        self.as_inner().is_titan()
     }
 
-    for cf in db.cf_names() {
-        delete_all_in_range_cf(db, cf, start_key, end_key, use_delete_range)?;
+    fn flush_cf(&self, cf: &str, sync: bool) -> Result<()> {
```

Comment:
Is interface `compact_cf` also need to be implemented by this trait?

Reply:
I'm not sure. I don't know what compact_cf is. I do have another patch after this one that moves a bunch of compaction methods into their own trait.

Reply:
TiKv may compact RocksDB by self. 
It is not very important and frequently to be called. Maybe we could add 'compact_cf' in other PR.